### PR TITLE
Make generated auth flows aware of captcha

### PR DESCRIPTION
**Pull request by pkong-ds**
## Discussions
1. [Resolved - `No`] Should allow captcha for reauth's identify step too? (ongoing discussion [here](https://github.com/authgear/authgear-server/pull/4282#discussion_r1641979288)) 
2. [Resolved - `Yes, but specify when consuming`]Should specify cascading relationship of root captcha (enabled) & branch captcha (required)? (ongoing discussion [here](https://github.com/authgear/authgear-server/pull/4282#discussion_r1641965018))
   - More test cases if necessary
3. Account recovery flow test case is missing - is it
    1. intentional; OR
    2. because JSON schema is dynamically generated? ref pkg/lib/config/authentication_flow.go:1438
4. Currently supports recursive definition of captcha; Assume only top layer is read

### CODE REVIEW

File: pkg/lib/config/testdata/login_flow_tests.yaml [hunk 1]
```diff
@@ -52,16 +58,20 @@ error: |-
...
     one_of:
     - identification: email
+      captcha:
+        foobar: true
       steps:
       - type: authenticate
         one_of:
         - authentication: foobar
-          target_step: 1
+          target_step: 1
```
Comment (louischan-oursky) [inline pkg/lib/config/testdata/login_flow_tests.yaml:74]: Remove extra spaces.

**Review by louischan-oursky — COMMENTED** [review]
- Make changes to pkg/lib/authenticationflow/declarative/generate_*.go, so that the generated flows respect the global captcha config. Refer to the spec for the generation behavior.

**Comment by louischan-oursky** [comment]
Please rebase.

**Comment by pkong-ds** [comment]
@louischan-oursky rebased, thanks!

**Comment by pkong-ds** [comment]
- seems `panic: test timed out after 10m0s`
- running `make -C e2e run` locally works fine

Suggest re-run ci, but I don't have access, can help press the button 🙏 ?

**Comment by tung2744** [comment]
It is a known issue that e2e tests will sometimes stuck. Re-running.

**Pull request merged**